### PR TITLE
feat: New module avm/res/synapse/private-link-hub

### PR DESCRIPTION
**Pull request by elanzel**
## Description

New module migrated from CARML to AVM.

Related to module proposal https://github.com/Azure/Azure-Verified-Modules/issues/610 

Status badge:
[avm.res.synapse.private-link-hub](https://github.com/elanzel/bicep-registry-modules/actions/workflows/avm.res.synapse.private-link-hub.yml)

### CODE REVIEW

File: avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        roleDefinitionIdOrName: 'Reader'
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        principalType: 'ServicePrincipal'
+      }
+      {
+        principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+        roleDefinitionIdOrName: '/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c'
+        principalType: 'ServicePrincipal'
+      }
+    ]
```
Comment (AlexanderSehr) [inline avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep:85]:
```suggestion
```
Requires update of docs

File: avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    lock: {
+      kind: 'CanNotDelete'
+      name: 'myCustomLockName'
+    }
```
Comment (AlexanderSehr) [inline avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep:59]:
```suggestion
```
Requires update of docs

File: avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    networkSecurityGroupName: 'dep-${namePrefix}-nsg-${serviceShort}'
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
```
Comment (AlexanderSehr) [inline avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep:40]: Not required, dependency can be removed

Comment (elanzel) [reply avm/res/synapse/private-link-hub/tests/e2e/waf-aligned/main.test.bicep]: managed identity removed

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**